### PR TITLE
fix(vue-app): multiple named views cause invalid syntax

### LINKED ISSUE #5096: multiple named views cause router.js contains syntax error
### Version

[v2.4.3](https://github.com/nuxt.js/releases/tag/v2.4.3)

### Reproduction link

[https://github.com/kdeng/nuxt-named-view-sample](https://github.com/kdeng/nuxt-named-view-sample)

### Steps to reproduce

1. Run `npm i`
2. Run `npm run dev`

### What is expected ?

I expected router.js can be generated properly, and named view can be rendered properly.

### What is actually happening?

router.js has been generated with syntax error.

```
 ERROR  Failed to compile with 1 errors                                                                                                                                                                                                                   friendly-errors 22:13:34


 ERROR  in ./.nuxt/router.js                                                                                                                                                                                                                              friendly-errors 22:13:34

Syntax Error: Unexpected token, expected "," (86:8)                                                                                                                                                                                                       friendly-errors 22:13:34

  84 |         default: _708facd7,
  85 |         footer: _03b372e3
> 86 |         default: _6fab42f0,
     |         ^
  87 |         top: _6b7085a2
  88 |       },
  89 |       name: "main"

                                                                                                                                                                                                                                                          friendly-errors 22:13:34
 @ ./.nuxt/index.js 8:0-43 59:19-31
 @ ./.nuxt/client.js
 @ multi eventsource-polyfill webpack-hot-middleware/client?reload=true&timeout=30000&ansiColors=&overlayStyles=&name=client&path=/__webpack_hmr/client ./.nuxt/client.js
                                                                                                                                                                                                                                                          friendly-errors 22:13:34
ℹ Waiting for file changes

```

### Additional comments?

I am new to nuxtjs, and try to play around about named views for enabling master footer component for cart router, I am not sure whether it is my mistake or a bug. Cheers.

<!--cmty--><!--cmty_prevent_hook-->
<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c8715">#c8715</a>)</em></sub></div>

### PR DESCRIPTION
**Pull request by leahciMic**
Fixes #5096

## Types of changes

- [x] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

* Reset resMap to an empty string before building the list of components
* Add another.vue which fails the build if the bug is present (as the
resulting .nuxt/router.js will contain a syntax error)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [x] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by robertpiosik** [comment]
Just faced this issue from the morning, trying to find solution and came across this PR. Simple and working solution, working as expected. Great work! 👏

I can't believe that named views was released in stable with such a bug 👀

**Comment by robertpiosik** [comment]
Just faced this issue from the morning, trying to find a solution and just came across this PR. Simple solution, working as expected 👏

I can't believe that named views was released in stable with such a bug 👀

As you changed "test/fixtures/named-views/nuxt.config.js" file using filter method on routes will remove those which not satisfy routes in passed function, what will break autogenerated router configuration. Am I wrong? Correct me if so

**Comment by leahciMic** [comment]
@robertpiosik 

> Faced this issue till the morning, trying to find out a solution and just came across this PR. Simple solution, working as expected :clap:

Thank you for confirming this fixes the issue!

> I can't believe that named views was released in stable with such a bug :eyes:

These things happen!

> As you changed "test/fixtures/named-views/nuxt.config.js" file; used filter method on routes will remove those which not satisfy ones in passed function, what breaks autogenerated router configuration. Am I wrong? Correct me if so

I believe you can return a new array from `extendRoutes` that will replace the routes, but in this case we only modify some of the objects in the array. `filter` itself does not mutate the array, so it will not drop any routes.

File: test/fixtures/named-views/nuxt.config.js [hunk 1]
```diff
@@ -1,30 +1,32 @@
 export default {
   router: {
     extendRoutes(routes, resolve) {
-      const indexIndex = routes.findIndex(route => route.name === 'index')
-      let index = routes[indexIndex].children.findIndex(route => route.name === 'index-child-id')
-      routes[indexIndex].children[index] = {
-        ...routes[indexIndex].children[index],
+      const indexChildRoute = routes.find(route => route.name === 'index')
+        .children.find(route => route.name === 'index-child-id')
```
Comment (manniL) [inline test/fixtures/named-views/nuxt.config.js]:
```suggestion
        .children.find(route => route.name === 'index-child-id')
```

could we extract that into a new variable? :relaxed:

File: test/fixtures/named-views/nuxt.config.js [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
-        components: {
-          default: routes[index].component,
-          top: resolve(__dirname, 'components/mainTop.vue')
-        },
-        chunkNames: {
-          top: 'components/mainTop'
-        }
-      }
+      routes
+        .filter(route => route.name === 'main' || route.name === 'another')
```
Comment (manniL) [inline test/fixtures/named-views/nuxt.config.js]:
```suggestion
        .filter(route => ['main', 'another'].includes(route.name))
```

**Review by TheAlexLichter — CHANGES_REQUESTED** [review]
Thanks so far! Left you some comments :relaxed:

File: test/fixtures/named-views/nuxt.config.js (hunk 1 above)
Comment (leahciMic) [reply test/fixtures/named-views/nuxt.config.js]: I agree, done!

File: test/fixtures/named-views/nuxt.config.js (hunk 2 above)
Comment (leahciMic) [reply test/fixtures/named-views/nuxt.config.js]: I agree, committed.

**Comment by leahciMic** [comment]
@manniL thank you for your review, I have addressed the feedback

**Review by TheAlexLichter — APPROVED** [review]
LGTM :relaxed:

**Comment by manniL** [comment]
https://github.com/nuxt/nuxt.js/issues/5096#issuecomment-474257336 <- :thinking:

**Comment by leahciMic** [comment]
@manniL I have looked at that comment, I believe there was some confusion as to how to test it. He has checked out my `dev` branch of nuxt which is very much out-of-date. I'm clearing up the instructions for him on how to test it.

**Pull request merged**